sdks/typescript/src/v1/client/client.ts

```diff
@@ -0,0 +1,151 @@
+  ClientConfig,
+  InternalHatchetClient,
+  HatchetClientOptions,
+} from '@hatchet/clients/hatchet-client';
+import { AxiosRequestConfig } from 'axios';
+import WorkflowRunRef from '@hatchet/util/workflow-run-ref';
+import { Workflow as V0Workflow } from '@hatchet/workflow';
+import { CreateWorkflow, CreateWorkflowOpts, RunOpts, Workflow } from '../workflow';
+import { IHatchetClient } from './client.interface';
+import { CreateWorkerOpts, Worker } from './worker';
+
+/**
+ * HatchetV1 implements the main client interface for interacting with the Hatchet workflow engine.
+ * It provides methods for creating and executing workflows, as well as managing workers.
+ */
+export class HatchetClient implements IHatchetClient {
+  /** The underlying v0 client instance */
+  v0: InternalHatchetClient;
+
+  /** The tenant ID for the Hatchet client */
+  get tenantId() {
+    return this.v0.tenantId;
+  }
+
+  /**
+   * Creates a new Hatchet client instance.
+   * @param config - Optional configuration for the client
+   * @param options - Optional client options
+   * @param axiosConfig - Optional Axios configuration for HTTP requests
+   */
+  constructor(
+    config?: Partial<ClientConfig>,
+    options?: HatchetClientOptions,
+    axiosConfig?: AxiosRequestConfig
+  ) {
+    this.v0 = new InternalHatchetClient(config, options, axiosConfig);
+  }
+
+  /**
+   * Static factory method to create a new Hatchet client instance.
+   * @param config - Optional configuration for the client
+   * @param options - Optional client options
+   * @param axiosConfig - Optional Axios configuration for HTTP requests
+   * @returns A new Hatchet client instance
+   */
+  static init(
+    config?: Partial<ClientConfig>,
+    options?: HatchetClientOptions,
+    axiosConfig?: AxiosRequestConfig
+  ): HatchetClient {
+    return new HatchetClient(config, options, axiosConfig);
+  }
+
+  /**
+   * Creates a new workflow definition.
+   * @template T - The input type for the workflow
+   * @template K - The return type of the workflow
+   * @param options - Configuration options for creating the workflow
+   * @returns A new Workflow instance
+   * @note It is possible to create an orphaned workflow if no client is available using @hatchet/client CreateWorkflow
+   */
+  workflow<T extends Record<string, any> = any, K = any>(
+    options: CreateWorkflowOpts
+  ): Workflow<T, K> {
+    return CreateWorkflow<T, K>(options, this);
+  }
+
+  /**
+   * Triggers a workflow run without waiting for completion.
+   * @template T - The input type for the workflow
+   * @template K - The return type of the workflow
+   * @param workflow - The workflow to run, either as a Workflow instance or workflow name
+   * @param input - The input data for the workflow
+   * @param options - Configuration options for the workflow run
+   * @returns A WorkflowRunRef containing the run ID and methods to interact with the run
+   */
+  enqueue<T extends Record<string, any> = any, K = any>(
```

Comment:
I think we should agree on naming for these across SDKs - want to call them `enqueue` and `run` in both cases? could there be any confusion there? I kinda like `run_and_wait` and `run_and_forget`, but open to ideas 🤷

Reply:
i personally like run and enqueue since they're less verbose. do you think people will be confused?

i dont like _and_wait here because both requests are async and will technically need to be awaited

Reply:
I think it's a little confusing but I don't have really strong feelings on naming 🤷 agreed on the `_and_wait` being confusing since `await` is its own thing. could use celery terminology with `run` and `delay` 😂 

I think `enqueue` and `run` is good - I'll do this on the Python side too so we can be consistent